core/src/epicli/data/common/ansible/playbooks/roles/repository/files/download-requirements/common/common_functions.sh

```diff
@@ -0,0 +1,122 @@
+echol() {
+#
+# Print to stdout, optionally to a log file.
+# Requires $CREATE_LOGFILE and $LOG_FILE_PATH to be defined.
+#
+# param $@: args which will be printed
+#
+    echo -e "$@"
+    if [[ $CREATE_LOGFILE == "yes" ]]; then
+        local timestamp=$(date +"%b %e %H:%M:%S")
+        echo -e "${timestamp}: $@" >> "$LOG_FILE_PATH"
+    fi
+}
+
+
+exit_with_error() {
+    echol $@
+    exit 1
+}
+
+
+__check_curl() {
+#
+# Use curl in silent mode to check if target `url` is available.
+#
+# param $1: url which will be tested
+#
+    [[ $# -lt 1 ]] && exit_with_error "(curl), no url provided"
+    local url=$1
+
+    echol "Testing connection to: ($url) using: (curl)..."
+
+    err_msg=$(curl -S --silent $url 2>&1 >/dev/null) || exit_with_error "(curl) failed, reason: ($err_msg)"
+}
+
+
+__check_wget() {
+#
+# Use wget in spider mode (without downloading resources) to check if target `url`
+# is available.
+#
+# param $1: url which will be tested
+#
+    [[ $# -lt 1 ]] && exit_with_error "(wget), no url provided"
+    local url=$1
+
+    echol "Testing connection to: ($url) using: (wget)..."
+
+    err_msg=$(wget --spider -nd $url 2>&1 >/dev/null) || exit_with_error "(wget) failed, reason: ($err_msg)"
```

Comment:
```suggestion
    err_msg=$(wget --spider --no-directories $url 2>&1 >/dev/null) || exit_with_error "wget failed, reason: ($err_msg)"
```
Do we need `--no-directories` option assuming that we will use only links to a single file?

I would prefer to return test result instead of exiting. In old Bash (as in RHEL 7) returning array is very limited but we could for example print array like: `printf '%s\n' "${array[@]}"` and have exit code in the first line - 0 - tests passed, 1 - failed. If failed, then have in next lines error message. Such approach would allow to run all tests in parent function then interpret results, for example print "report"/summary for all tools.